### ISSUE TITLE
Perspective change in 3d gives visual artefacts for points layer

### ISSUE THREAD

**Issue by alisterburt** [napari/napari#2637]
to reproduce: add points, view in 3d then shift, right click and drag

![image](https://user-images.githubusercontent.com/7307488/116629506-5218dd00-a949-11eb-8c7e-a36cca466955.png)

```
napari: 0.4.8.dev36+g88db2f06.d20210418
Platform: macOS-11.2.1-x86_64-i386-64bit
System: MacOS 11.2.1
Python: 3.9.2 | packaged by conda-forge | (default, Feb 21 2021, 05:02:20)  [Clang 11.0.1 ]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.20.2
SciPy: 1.6.2
Dask: 2021.04.0
VisPy: 0.6.6

OpenGL:
  - GL version:  2.1 Metal - 71.0.7
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1920x1080, scale 1.0
  - screen 2: resolution 1440x900, scale 2.0

Plugins:
  - console: 0.0.3
  - scikit-image: 0.4.8.dev36+g88db2f06.d20210418
  - svg: 0.1.4
```

**Comment by sofroniewn** [comment]
It looks like it is just the size of the points that changes, possibly because something here is no longer quite right https://github.com/vispy/vispy/blob/e33aae23847e8e5f43248ba9d014cc4aea211a90/vispy/visuals/markers.py#L623-L626

**Comment by kevinyamauchi** [comment]
I think I have come up with a fix for this: https://github.com/vispy/vispy/pull/2198

If the vispy folks agree, we can vendor the shader and implement it here.

**Comment by sofroniewn** [comment]
Nice @kevinyamauchi I just flagged https://github.com/napari/napari/issues/2894#issuecomment-906792080 for you too - not sure if it is related, but might be good for you to look at too

**Comment by kevinyamauchi** [comment]
I'll have a look!

**Comment by brisvag** [comment]
Closed by #3494.
